### ISSUE TITLE
How I can load non-lazy component in lazy component and viceversa

### ISSUE THREAD

**Issue by progressify** [sis0k0/lazyNinjas#15]
Hi, I have created a lazy component and seems to work well
but...
In the XML of this component, I use other 2 custom component non-lazy, and there are not shown in the "activity" (I have tested only on android)

Have you some idea?

**Comment by sis0k0** [comment]
Hey @progressify, did you register the non-lazy components as routes as well? If the answer is yes - can you share your routing configuration?

**Comment by progressify** [comment]
No, some of the non-lazy components don't have a route, it was simply utils components
I must insert a route also for utils components?

**Comment by progressify** [comment]
this is an example:
```
<DockLayout>
            <header-menu></header-menu>

            <footer-menu></footer-menu>
            
            <StackLayout class="content">
                             .......
                             .......
            </StackLayout>
</DockLayout>
```

header-menu and footer-menu are a non-lazy component but it has not the route because are only utils component.

**Comment by sis0k0** [comment]
@progressify, which NgModule declares the `header-menu` and `footer-menu` components?

**Comment by progressify** [comment]
in the app.module I have inserted these components

this is my app.module

```
@NgModule({
    bootstrap: [
        AppComponent
    ],
    imports: [
        NativeScriptModule,
        NativeScriptFormsModule,
        AppRoutingModule,
        HttpClientModule,
        NativeScriptUIAutoCompleteTextViewModule,
        NativeScriptUISideDrawerModule,
        TNSCheckBoxModule,
    ],
    declarations: [
        AppComponent,
        ...
        ...
        HeaderMenuComponent,
        FooterMenuComponent,
        ...
        ...
    ],
    providers: [
        AppService,
        ...
        ...
    ],
    schemas: [
        NO_ERRORS_SCHEMA
    ]
})
export class AppModule { }
```

**Comment by sis0k0** [comment]
@progressify, currently, the components are declared in the root NgModule - AppModule. That's why the lazily loaded module cannot use them. You may have to move these components to a shared NgModule and import the latter everywhere you have to need to use the shared components. I made a simple sample web app - https://github.com/sis0k0/shared-module-sample. The concept is the same for {N}.

**Comment by progressify** [comment]
thanks, this has solved my problem.
Today I have refactored all code, but I have other problems now.
I have created a component for Drawer (I use nativescript-ui-sidedrawer: RadSideDrawer) if I move this component in SharedModule it not work.

Another problem is: in lazy component how I can use another lazy component?

**Comment by sis0k0** [comment]
@progressify,

1. The RadSideDrawer is usually used as a root element and don't need to be shared. Check out this template - https://github.com/NativeScript/template-drawer-navigation-ng/blob/master/src/app/app.component.html.

2. Using lazy component in another lazy component.
I'm afraid I couldn't understand your question here. Could you give me an example?

**Comment by progressify** [comment]
Hi, sorry for delay and thanks for your help and your patience

Thanks for the first point. I have refactored the code of the RadSideDrawer. 
For the second point, I give you an example:

CompOne -> a lazy component
CompTwo -> another lazy component

in comp_one.component.ts or in comp_one.component.html I have a router link redirection to CompTwo.
```
this.routerExtensions.navigate(["/comp_two/param1"], { clearHistory: true });
```
or
```
[nsRouterLink]="['/comp_two/param1]"
```
but it not work, when I tap on the element and I have no error in console

EDIT: another problem :cry: 
after the refactor of the code, for a drawer in app.component, apparently two-way binding not work

**Comment by sis0k0** [comment]
Hey again!
Can you share the error message that you get when you try to navigate?
For the `SideDrawer` problem - please, try to refer to the template from the link above. If it doesn't help, can you send me a short reproduction of the problem? If it's not an option to attach it here - feel free to send it to stanimira.vlaeva@gmail.com.

**Comment by progressify** [comment]
hi @sis0k0 
yes, I have followed the template you have sent me.
this is an example of that I have implemented

[LazyDrawerTest.zip](https://github.com/sis0k0/lazyNinjas/files/2824929/LazyDrawerTest.zip)

In the `console.log("login", this.inputs);` I think to find the data I have inserted in the TextField, but i receive this:
```
login {
JS:   "email": "",
JS:   "password": ""
JS: }
```

without lazy-loading it works, with lazy-loading it does not work :((

**Comment by progressify** [comment]
hi @sis0k0, I have resolved this problem.
[This issue](https://github.com/NativeScript/nativescript-angular/issues/735) have the solution!

My shared module now have this 3 imports/exports 

```
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { NativeScriptFormsModule } from 'nativescript-angular/forms';
import { ReactiveFormsModule } from '@angular/forms';
import { NativeScriptCommonModule } from '../../node_modules/nativescript-angular/common';

@NgModule({
  imports: [
    NativeScriptCommonModule,
    NativeScriptFormsModule,
    ReactiveFormsModule,
    ....
    ....
  ],
  exports: [
    NativeScriptCommonModule,
    NativeScriptFormsModule,
    ReactiveFormsModule,
    ....
    ....
  ],
  declarations: [],
  schemas: [
    NO_ERRORS_SCHEMA
  ]
})
export class SharedModule { }
```

For the other problem, I don't have any news :(
If I find the solution I write it on this issue ;)
